hilti/toolchain/src/compiler/codegen/ctors.cc

```diff
@@ -116,7 +116,13 @@ struct Visitor : hilti::visitor::PreOrder {
                                                     }),
                                     " ");
 
-            result = fmt("[&]() { auto __xs = ::hilti::rt::Map<%s, %s>(); %s return __xs; }()", k, v, elems);
+            // If we are at block scope capture other variables so they can be
```

Comment:
One caveat with not capturing is that we rely on any constants referenced by the declaration to have be declared before (since we emit C++ `const`: actually initialized), e.g., code like this would work as expected:

```ruby
const a = 1;
const b = vector(a, ...);
```

In general the expectation in Spicy/HILTI is that order of declarations does not matter, but this currently does not work for `const` variables, see #1763. That is unfortunate, but luckily in this case means that an empty capture `[]` should be sufficient.